functional_tests/test_multiple_users_multiple_channels_multiple_revisions.py

```diff
@@ -0,0 +1,102 @@
+import time
+import pytest
+from lib.user import User
+import concurrent.futures
+from lib.admin import Admin
+from fixtures import cluster
+import pytest
+from lib.parallelize import *
+import logging
+log = logging.getLogger(settings.LOGGER)
+
+
+
+# Scenario-2:
+# Single User Single Channel: Create Unique docs and update docs verify all num docs present in changes feed.
+# Verify all revisions in changes feed
+# https://docs.google.com/spreadsheets/d/1nlba3SsWagDrnAep3rDZHXHIDmRH_FFDeTaYJms_55k/edit#gid=598127796
+
+@pytest.mark.sanity
+@pytest.mark.distributed_index
+@pytest.mark.parametrize("num_users", [5])
+@pytest.mark.parametrize("num_channels", [3]) #all users share all channels
+@pytest.mark.parametrize("num_docs", [2])
+@pytest.mark.parametrize("num_revisions", [10])
+def test_mulitple_users_mulitiple_channels_mulitple_revisions(cluster, num_users,num_channels, num_docs, num_revisions):
+
+    log.info("Starting test...")
+    start = time.time()
+
+    cluster.reset(config="sync_gateway_default_functional_tests.json")
+
+    init_completed = time.time()
+    log.info("Initialization completed. Time taken:{}s".format(init_completed - start))
+
+    users = ["User-" + str(i) for i in range(num_users)]
+    channels = ["channel-" + str(i) for i in range(num_channels)]
+    password = "password"
+    user_objects = []
+    use_uuid_names = True
+
+    sgs = cluster.sync_gateways
+
+    admin = Admin(sgs[0])
+
+    # Register User
+    log.info("Register User")
+    for username in users:
```

Comment:
You can use admin.register_bulk_users(self, target, db, name_prefix, number, password, channels=list(), roles=list())